sqlalchemy_exasol/base.py

```diff
@@ -802,19 +835,30 @@ def get_pk_constraint_odbc(self, connection, odbc_connection, table_name, schema
     @reflection.cache
     def get_pk_constraint_sql(self, connection, table_name, schema=None, **kw):
         schema = self._get_schema_for_input(connection, schema)
+        table_name = self.denormalize_name(table_name)
+        table_name_string = ":table"
+        if schema is None:
+            schema_string = "CURRENT_SCHEMA "
+        else:
+            schema_string = ":schema "
+        sql_stmnt=self._get_constraint_sql_str(schema_string, table_name_string, "PRIMARY KEY")
+        rp = connection.execute(sql.text(sql_stmnt),
+                                schema=self.denormalize_name(schema),
+                                table=table_name)
         pkeys = []
         constraint_name = None
-        table_name = self.denormalize_name(table_name)
-
-        for row in self._get_all_constraints(connection, schema, info_cache=kw.get("info_cache")):
+        for row in list(rp):
             if (row[5] != table_name and table_name is not None) or row[6] != 'PRIMARY KEY':
                 continue
             pkeys.append(self.normalize_name(row[1]))
             constraint_name = self.normalize_name(row[0])
         return {'constrained_columns': pkeys, 'name': constraint_name}
 
+
     @reflection.cache
     def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        if table_name is None:
+            raise RuntimeError("table_name must not null")
```

Comment:
Was this previously an uncaught exception or the exception didn't happen in the first place?

Reply:
Previously, we returned primary keys for all tables, however the interface documentation of sqlalchemy doesn't describe the case. As an alternative, we could return None or something like that.

  https://docs.sqlalchemy.org/en/13/core/internals.html?highlight=get_columns#sqlalchemy.engine.interfaces.Dialect.get_pk_constraint